cluster/examples/kubernetes/ceph/create-external-cluster-resources.py

```diff
@@ -0,0 +1,633 @@
+import json
+import argparse
+import unittest
+import re
+from os import linesep as LINESEP
+
+# backward compatibility with 2.x
+try:
+    ModuleNotFoundError
+except:
+    ModuleNotFoundError = ImportError
+
+try:
+    import rados
+except ModuleNotFoundError as noModErr:
+    print("Error: %s\nExiting the script..." % noModErr)
+    sys.exit(1)
+
+try:
+    # for 2.7.x
+    from StringIO import StringIO
+except ModuleNotFoundError:
+    # for 3.x
+    from io import StringIO
+
+
+class ExecutionFailureException(Exception):
+    pass
+
+
+class RadosJSON:
+    EXTERNAL_USER_NAME = "client.healthchecker"
+    EMPTY_OUTPUT_LIST = "Empty output list"
+
+    @classmethod
+    def gen_arg_parser(cls, args_to_parse=None):
+        argP = argparse.ArgumentParser()
+        argP.add_argument("--verbose", "-v",
+                          action='store_true', default=False)
+        argP.add_argument("--ceph-conf", "-c",
+                          help="Provide a ceph conf file.", type=str)
+        argP.add_argument("--run-as-user", "-u",
+                          help="Provides a user name to check the cluster's health status",
+                          default=cls.EXTERNAL_USER_NAME, type=str)
+        argP.add_argument("--format", "-t", choices=["json", "bash"],
+                          default='json', help="Provides the output format (json | bash)")
+        argP.add_argument("--cluster-name", default="openshift-storage",
+                          help="Ceph cluster name")
+        argP.add_argument("--output", "-o", default="",
+                          help="Output will be stored into the provided file")
+        argP.add_argument("--cephfs-filesystem-name", default="",
+                          help="Provides the name of the Ceph filesystem")
+        argP.add_argument("--cephfs-data-pool-name", default="",
+                          help="Provides the name of the cephfs data pool")
+        argP.add_argument("--rbd-data-pool-name", default="", required=True,
+                          help="Provides the name of the RBD datapool")
+        argP.add_argument("--namespace", default="",
+                          help="Namespace where CephCluster is running")
+        argP.add_argument("--rgw-pool-prefix", default="default",
+                          help="RGW Pool prefix")
+        argP.add_argument("--rgw-endpoint", default="", required=True,
+                          help="Rados GateWay endpoint (in <IP>:<PORT> format)")
+        if args_to_parse:
+            assert type(args_to_parse) == list, \
+                "Argument to 'gen_arg_parser' should be a list"
+        else:
+            args_to_parse = sys.argv[1:]
+        return argP.parse_args(args_to_parse)
+
+    def _invalid_endpoint(self, endpoint_str):
+        try:
+            ipv4, port = endpoint_str.split(':')
+        except ValueError:
+            raise ExecutionFailureException(
+                "Not a proper endpoint: {}, multiple ':' found".format(endpoint_str))
```

Comment:
I think the `multiple ':' found` is confusing since it will be displayed even when no `:` are given. Let's just remind what's the expected format --> IP:PORT.

Reply:
Yes, you are right... Changed it...